File: src/main/java/be/breina/openrgb/model/ObjectLinker.java
package be.breina.openrgb.model;

import be.breina.openrgb.OpenRgbClient;

public final class ObjectLinker {

    public static void link(Device device, int deviceIndex, OpenRgbClient client) {
        device.colorMutator(colors -> client.updateLeds(deviceIndex, colors));

        Led[] leds = device.getLeds();
        for (int i = 0, ledsLength = leds.length; i < ledsLength; i++) {
            Led led = leds[i];

            final int ledIndex = i;
            led.colorMutator(color -> {
                client.updateLed(deviceIndex, ledIndex, color);
                device.getColors()[ledIndex].setColor(color);
            });

        }
    }
}

File: src/main/kotlin/be/breina/openrgb/model/ObjectLinker.kt
package be.breina.openrgb.model

import be.breina.openrgb.OpenRgbClient
import java.awt.Color

object ObjectLinker {
    fun link(device: Device?, deviceIndex: Int, client: OpenRgbClient) {
        device!!.colorMutator { colors: Array<Color> -> client.updateLeds(deviceIndex, *colors) }
        val leds = device.leds
        var i = 0
        val ledsLength = leds.size
        while (i < ledsLength) {
            val led = leds[i]
            val ledIndex = i
            led.colorMutator { color: Color ->
                client.updateLed(deviceIndex, ledIndex, color)
                val c = device.colors[ledIndex]
                c.red = color.red.toByte()
                c.green = color.green.toByte()
                c.blue = color.blue.toByte()
            }
            i++
        }
    }
}